contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -309,6 +312,8 @@ public VoldemortBuildAndPushJob(String name, azkaban.utils.Props azkabanProps) {
         }
         this.executorService = Executors.newFixedThreadPool(requiredNumberOfThreads);
 
+        this.storeVerificationExecutorService = Executors.newFixedThreadPool(THREAD_NUM_FOR_STORE_VERIFICATION);
```

Comment:
In my opinion, there should be a method called preProcess before runBuildStore, which validates the store.

This method creates the thread pool and in the finally block, closes it. In the future, if the thread pool is required at other places, we can extend the scope, but to start with we can make it as a limited scope.